steps/circuit.py

```diff
@@ -0,0 +1,36 @@
+import numpy as np
+from zquantum.core.circuit import save_circuit, Circuit, load_circuit_template_params
+from zquantum.core.openfermion import load_qubit_operator
+from zquantum.core.utils import create_object, load_from_specs
+from typing import Union, List, Optional, Dict
+
+Specs = Union[str, Dict]
+
+
+def build_qaoa_ansatz_circuit(
+    ansatz_specs: Specs,
+    cost_hamiltonian: Union[str, List],
+    mixer_hamiltonian: Union[str, List] = None,
+    params: Optional[Union[str, List]] = None,
+):
+    cost_hamiltonian = load_qubit_operator(cost_hamiltonian)
+    if mixer_hamiltonian:
+        mixer_hamiltonian = load_qubit_operator(mixer_hamiltonian)
+    ansatz_specs["cost_hamiltonian"] = cost_hamiltonian
+    ansatz_specs["mixer_hamiltonian"] = mixer_hamiltonian
```

Comment:
So this is a bit of a strange edge case. Since `ansatz_specs` can be either a string or a dict, this will not work unless it's a dict.

I see a few options to proceed:
1. Have a conditional to check the type of `ansatz_specs` and modify `ansatz_specs` appropriately depending on the case.
2. Change `ansatz_specs` to a `dict` at the beginning of the function if it's a string
3. Change type of `ansatz_specs` to only allow `dict`s

I think, long-term, (3) is the best option here since we should be deprecating the "string support" for specs now that we can use dictionaries in the workflow, but this will cause headaches in the meantime.

Reply:
What about this for now:
https://github.com/zapatacomputing/z-quantum-qaoa/pull/27/commits/25d4b6cb62b4aee61e1a8af9dabe7c4922b35ce3
?

Reply:
I think it's fair